### PR TITLE
Fixed icon markup

### PR DESCRIPTION
**Pull request by zoephix**

### CODE REVIEW

**Comment by Heyter** [comment]
https://wiki.facepunch.com/gmod/table.getn

File: gamemode/core/libs/cl_markup.lua [hunk 1]
```diff
@@ -22,444 +22,490 @@ TEXT_ALIGN_TOP		= 3
...
-
+		
 		local text_block = {}
-		text_block.text = p
-		text_block.colour = colour_stack[ #colour_stack ]
-		text_block.font = font_stack[ #font_stack ]
-		table.insert( blocks, text_block )
-
+		text_block.text = p 
+		text_block.colour = colour_stack[ table.getn(colour_stack) ]
```
Comment (TovarischPootis) [inline gamemode/core/libs/cl_markup.lua]:
```suggestion
		text_block.colour = colour_stack[ table.Count(colour_stack) ]
```

File: gamemode/core/libs/cl_markup.lua [hunk 2]
```diff
@@ -22,444 +22,490 @@ TEXT_ALIGN_TOP		= 3
...
+		text_block.font = font_stack[ table.getn(font_stack) ]
```
Comment (TovarischPootis) [inline gamemode/core/libs/cl_markup.lua]:
```suggestion
		text_block.font = font_stack[ table.Count(font_stack) ]
```

**Review by TovarischPootis — CHANGES_REQUESTED** [review]
getn is depricated, use Count for non-sequential tables instead

**Comment by Heyter** [comment]
table.Count slower than "operator #"
"#colour_stack"

**Comment by TovarischPootis** [comment]
> table.Count slower than "operator #"
> "#colour_stack"

The tables in question are non-sequential ({r = #, g = #, b = #, a = #}).
`#` wouldn’t work, hence table.Count is needed instead

**Comment by Heyter** [comment]
> > table.Count slower than "operator #"
> > "#colour_stack"
> 
> The tables in question are non-sequential ({r = #, g = #, b = #, a = #}).
> `#` wouldn’t work, hence table.Count is needed instead

Is this prank?

```
local colour_stack = { {r=255,g=255,b=255,a=255} }
table.insert(colour_stack, {r=255,g=255,b=255,a=255})
print(#colour_stack)
```

Output: 2

**Comment by TovarischPootis** [comment]
You’re right, I was looking at the wrong thing. # works better here. My bad

File: gamemode/core/libs/cl_markup.lua [hunk 3]
```diff
@@ -22,444 +22,490 @@ TEXT_ALIGN_TOP		= 3
...
-
+		
 		local text_block = {}
-		text_block.text = p
-		text_block.colour = colour_stack[ #colour_stack ]
-		text_block.font = font_stack[ #font_stack ]
-		table.insert( blocks, text_block )
-
+		text_block.text = p 
+		text_block.colour = colour_stack[ table.Count(colour_stack) ]
```
Comment (TovarischPootis) [inline gamemode/core/libs/cl_markup.lua]:
```suggestion
		text_block.colour = colour_stack[#colour_stack ]
```
As pointed out by Heyter, # is even better here

File: gamemode/core/libs/cl_markup.lua [hunk 4]
```diff
@@ -22,444 +22,490 @@ TEXT_ALIGN_TOP		= 3
...
+		text_block.font = font_stack[ table.Count(font_stack) ]
```
Comment (TovarischPootis) [inline gamemode/core/libs/cl_markup.lua]:
```suggestion
		text_block.font = font_stack[ #font_stack ]
```
As pointed out by Heyter, # is even better here

**Review by TovarischPootis — CHANGES_REQUESTED** [review]

**Comment by TovarischPootis** [comment]
This is fine now, fixes an ancient bug with NS

**Pull request merged**